monai_ex/transforms/spatial/array.py

```diff
@@ -319,3 +323,71 @@ def __call__(self, img: NdarrayOrTensor, affine: Optional[np.ndarray] = None):
         if self.image_only:
             return output_data
         return output_data, affine, new_affine
+
+
+class RandomDrop(Randomizable, Transform):
+    """ Drop a number of patchs on the centerline of curvilinear structure randomly.
+
+    Args:
+        roi_number (int): Number of the drop roi.
+        roi_size (int): size of the drop ROI.
+        random_seed (int) : set the random state with an integer seed.
+    """
+    def __init__(
+        self,
+        roi_number: int,
+        roi_size: int,
+        random_state: int = 20221201
+    ) -> None:
+        self.roi_num = roi_number
+        self.roi_size = roi_size
+        self.random_seed = random_state
+
+    def __call__(
+        self,
+        img: NdarrayOrTensor,
+        cline: NdarrayOrTensor
+    ) -> NdarrayOrTensor:
+        """
+            Args:
+            img (NdarrayOrTensor): The image to drop patches.
+            cline (NdarrayOrTensor): The centerline of curvilinear structure, it can be extracted with `monai_ex.transforms.ExtractCenterline`
+        """
+        data = deepcopy(img)
+        dropped_data = []
+        c_points = np.nonzero(cline)
+        random.seed(self.random_seed)
```

Comment:
You dont need random yourself. Refer to monai's [random transform](https://github.com/Project-MONAI/MONAI/blob/0c2fbac6eb55a546886921d9add66be9688d9775/monai/transforms/croppad/array.py#L601). Put your randomizable part into a separate `def randomize(self)` function and use `self.R.sample` or `self.R.shuffle` to do random.